code_search/app/code_search/t2t/query.py

```diff
@@ -0,0 +1,32 @@
+import base64
+import tensorflow as tf
+from tensor2tensor.data_generators import text_encoder
+from tensor2tensor.utils import registry
+
+
+def get_encoder_decoder(problem_name, data_dir):
+  """Get encoder from the T2T problem.This might
+  vary by problem, keeping generic as a reference
+  """
+  problem = registry.problem(problem_name)
+  hparams = tf.contrib.training.HParams(data_dir=data_dir)
+  problem.get_hparams(hparams)
+  return problem.feature_info["inputs"].encoder, \
+         problem.feature_info["targets"].encoder
+
+
+def encode_query(encoder, query_str):
+  """Encode the input query string using encoder. This
+  might vary by problem but keeping generic as a reference.
+  Note that in T2T problems, the 'targets' key is needed
+  even though it is ignored during inference.
+  See tensorflow/tensor2tensor#868"""
+
+  encoded_str = encoder.encode(query_str) + [text_encoder.EOS_ID]
```

Comment:
If you're calling encode here, then what purpose is the model being served by TF Serving serving?

Reply:
Great point! I had similar concerns when starting but turns out TF Serving can itself not do anything with the incoming `instances` list, whether REST or gRPC, unless there is some part of the graph that does this transform. The sample model that I am using with this unfortunately doesn't have that part, so I had to manually do it. My next plan is to figure out the best way to write a T2T model which can handle all these things so that during export it also has such transformer functions. 

Consider this a work in progress but if nothing works out, this at least gets the inference done meanwhile.